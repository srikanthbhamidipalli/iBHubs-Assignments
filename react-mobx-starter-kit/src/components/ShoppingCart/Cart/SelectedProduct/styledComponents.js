import styled from "styled-components";

export const EachCartItem = styled.div`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 2px;
  box-shadow: -4px -3px 5px 1px #404040;
`;

export const ProductInfoContainer = styled.div`
  display: flex;
  flex-basis: 100%;
  flex-direction: column;
  padding-left: 15px;
`;

export const SizeAndQuantityInfo = styled.span`
  color: grey;
`;

export const PriceInsideCart = styled.span`
  color: orange;
`;

export const DeleteIcon = styled.span`
  position: absolute;
  font-weight: 600;
  color: white;
  right: 2px;
  top: 6px;
  cursor: pointer;
  ${props =>
    props.hoverStatus
      ? EachCartItem
      : "hover" &
        {
          "text-decoration": "line-through"
        }}
`;

export const ProductPicInsideCart = styled.img`
  height: 150px;
  width: 70px;
`;
