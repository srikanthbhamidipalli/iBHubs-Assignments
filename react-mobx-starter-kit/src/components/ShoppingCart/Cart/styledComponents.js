import styled from "styled-components";
export const InsideCartIconDiv = styled.div`
  height: 102px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;
export const SideNavBarBody = styled.div`
  background-color: rgb(8, 7, 7);
  width: 30vw;
  color: white;
`;
export const OutsideDivWithIcon = styled.div`
  position: relative;
`;
export const OutsideCartDiv = styled.div`
  height: 60px;
  width: 60px;
  cursor: pointer;
`;
export const SideNavBar = styled.div`
  position: fixed;
  right: ${props => (props.clickedStatus ? "0vw" : "-30vw")};
  display: flex;
  height: 100vh;
  transition: 0.9s ease-in-out;
`;
export const CheckOut = styled.div`
  height: 46px;
  width: 272px;
  background-color: black;
  color: white;
  margin-left: 116px;
`;

export const OutsideCartIcon = styled.img`
  width: 60px;
  height: 60px;
`;

export const OutsideCartItemsCount = styled.div`
  position: absolute;
  right: 11px;
  top: 30px;
  height: 20px;
  width: 20px;
  border-radius: 11px;
  background-color: yellow;
  color: black;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

export const OutsideCartDivision = styled.div`
  position: fixed;
  cursor: pointer;
`;

export const CrossIcon = styled.div`
  width: 60px;
  height: 60px;
  background-color: rgb(8, 7, 7);
  color: white;
  font-size: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CartInfoInsideNavBar = styled.div`
  height: 102px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

export const InsideCartIcon = styled.img`
  width: 60px;
  height: 60px;
`;

export const InsideCartItemsCount = styled.div`
  margin-top: 15px;
  height: 20px;
  width: 20px;
  border-radius: 11px;
  background-color: yellow;
  color: black;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  position: absolute;
`;

export const CartItemsContainer = styled.div`
  margin-top: 60px;
  height: 580px;
  background-color: transparent;
  border: 1px solid;
  padding: 10px;
  box-shadow: 5px 5px 8px #888888;
`;

export const SubTotalContainer = styled.div`
  height: 53px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
`;

export const AddToCartText = styled.span`
  height: 46px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: pointer;
`;
