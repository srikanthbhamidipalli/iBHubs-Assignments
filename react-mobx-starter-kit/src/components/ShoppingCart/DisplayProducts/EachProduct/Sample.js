import React, { Component } from "react";
class Sample extends Component {
  state = {};
  showMessage = () => {
    this.props.showMessage();
  };
  render() {
    return <div onClick={this.showMessage} />;
  }
}

export default Sample;
