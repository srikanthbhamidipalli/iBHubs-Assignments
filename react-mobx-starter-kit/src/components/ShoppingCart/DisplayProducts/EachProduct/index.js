import React, { Component } from "react";
import { observer } from "mobx-react";
import { observable } from "mobx";

import {
  EachProductDivision,
  ProductImage,
  FreeShippingLabel,
  FreeShippingText,
  ColouredDash,
  SizeSelector,
  AddToCart,
  AddToCartText
} from "./styledComponents";
import { orderByOptions, alertMessageForSizeSelection } from "../../constants";

@observer
class EachProduct extends Component {
  @observable selectedSize = orderByOptions.select;
  handleClick = () => {
    this.selectedSize === orderByOptions.select
      ? alert(alertMessageForSizeSelection)
      : this.props.addProductToCart(this.props.eachItem.id, this.selectedSize);
  };

  handleSizeClick = e => {
    this.selectedSize = e.target.value;
  };
  render() {
    const { eachItem } = this.props;
    let selectSizeList = [];
    const select = (
      <option key={orderByOptions.select} value={orderByOptions.select}>
        select Size
      </option>
    );
    selectSizeList.push(select);
    for (let i = 0; i < eachItem.availableSizes.length; i++) {
      const eachSize = (
        <option
          key={eachItem.availableSizes[i]}
          value={eachItem.availableSizes[i]}
        >
          {eachItem.availableSizes[i]}
        </option>
      );
      selectSizeList.push(eachSize);
    }
    return (
      <EachProductDivision>
        <ProductImage>
          <img src={eachItem.image} alt="productImage" />
          {eachItem.isFreeShipping ? (
            <FreeShippingLabel>
              <FreeShippingText>Free Shipping</FreeShippingText>
            </FreeShippingLabel>
          ) : null}
        </ProductImage>
        <br />
        <div>{eachItem.title}</div>
        <br />
        <ColouredDash />
        <br />
        <SizeSelector>
          <select onClick={this.handleSizeClick}>{selectSizeList}</select>
        </SizeSelector>
        <div id="price">${eachItem.price}</div>
        <div>
          or{" "}
          {eachItem.installments +
            " *" +
            (eachItem.price / eachItem.installments).toFixed(2)}
        </div>
        <br />
        <AddToCart onClick={this.handleClick}>
          <AddToCartText>Add to Cart</AddToCartText>
        </AddToCart>
      </EachProductDivision>
    );
  }
}

export default EachProduct;
