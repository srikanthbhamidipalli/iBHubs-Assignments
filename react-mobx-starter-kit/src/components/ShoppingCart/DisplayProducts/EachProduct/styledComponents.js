import styled from "styled-components";

export const EachProductDivision = styled.div`
  margin-top: 3px;
  height: 711px;
  width: 379px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  :hover {
    cursor: pointer;
    border: 1px solid grey;
  }
`;

export const ProductImage = styled.div`
  position: relative;
  width: 339px;
`;

export const FreeShippingLabel = styled.div`
  position: absolute;
  top: 0px;
  right: -10px;
  height: 23px;
  width: 103px;
  background-color: black;
  color: white;
  font-size: 13px;
  margin-left: 223px;
`;

export const FreeShippingText = styled.span`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 23px;
`;

export const ColouredDash = styled.div`
  background-color: orange;
  width: 27px;
  height: 3px;
`;

export const SizeSelector = styled.div``;

export const AddToCart = styled.div`
  height: 46px;
  width: 272px;
  background-color: black;
  color: white;
  margin-bottom: 8px;
  ${EachProductDivision}:hover & {
    background-color: orange;
  }
`;

export const AddToCartText = styled.span`
  height: 46px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;
