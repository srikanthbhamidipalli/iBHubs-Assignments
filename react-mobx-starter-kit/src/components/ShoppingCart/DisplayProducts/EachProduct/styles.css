.each-product-div {
  margin-top: 3px;
  height: 711px;
  width: 379px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.free-shipping-label {
  position: absolute;
  top: 0px;
  right: -10px;
  height: 23px;
  width: 103px;
  background-color: black;
  color: white;
  font-size: 13px;
  margin-left: 223px;
}

.free-shipping-text {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 23px;
}

.add-to-cart {
  height: 46px;
  width: 272px;
  background-color: black;
  color: white;
  margin-bottom: 8px;
}

.each-product-div:hover .add-to-cart {
  background-color: orange;
}

.each-product-div:hover {
  cursor: pointer;
  border: 1px solid grey;
}

.add-to-cart-text {
  height: 46px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.product-img {
  position: relative;
  width: 339px;
}

.color-dash {
  background-color: orange;
  width: 27px;
  height: 3px;
}
