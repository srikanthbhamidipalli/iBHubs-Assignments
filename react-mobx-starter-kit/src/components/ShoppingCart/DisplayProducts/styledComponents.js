import styled from "styled-components";

export const AllProductsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const PageStatus = styled.div`
  padding-left: 600px;
  height: 500px;
  width: 500px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
