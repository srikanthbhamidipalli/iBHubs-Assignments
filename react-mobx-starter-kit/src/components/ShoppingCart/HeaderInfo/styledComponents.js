import styled from "styled-components";

export const HeaderContent = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 1520px;
  height: 30px;
`;

export const SelectList = styled.select`
  border: 2px solid #bbb;
  :hover {
    border-color: black;
  }
`;

export const Option = styled.option``;

export const Span = styled.span``;
