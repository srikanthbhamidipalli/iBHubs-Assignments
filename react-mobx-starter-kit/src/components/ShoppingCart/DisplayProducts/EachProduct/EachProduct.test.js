import React from "react";
import { mount, shallow, render } from "enzyme";

import { appStore } from "../../../../AppStore";
import EachProduct from "./index";
import { FreeShippingLabel, AddToCart } from "./styledComponents";

const product = {
  availableSizes: ["S", "XS"],
  currencyFormat: "$",
  currencyId: "USD",
  description: "4 MSL",
  id: 12,
  installments: 9,
  isFreeShipping: true,
  price: 10.9,
  sku: 12064273040195392,
  style: "Black with custom print",
  title: "Cat Tee Black T-Shirt",
  image:
    "https://react-shopping-cart-67954.firebaseapp.com/static/media/12064273040195392_1.2995d79a.jpg"
};
describe("<EachProduct/>", () => {
  it("component Exists?", () => {
    const callBack = jest.fn();
    const wrapper = mount(
      <EachProduct
        key="2"
        appStore={appStore}
        eachItem={product}
        addProductToCart={callBack}
      />
    );
    // wrapper.find(AddToCart).simulate("click");
    // expect(callBack).toHaveBeenCalled();
    const price = wrapper.find("div#price").text();
    expect(price).toEqual("$10.9");
    expect(wrapper.find(FreeShippingLabel).exists()).toBeTruthy();
    expect();
  });
});

it("Select your Size dropdown is working or not?", () => {
  let callBack = jest.fn();
  const spy = jest.spyOn(appStore, "addProductToCart");
  const wrapper = mount(
    <EachProduct
      key="2"
      appStore={appStore}
      eachItem={product}
      addProductToCart={appStore.addProductToCart}
    />
  );
  wrapper
    .find("select#select-list")
    .simulate("click", { target: { value: "S" } });
  //AddtoCart handle click is working or not
  wrapper.find(AddToCart).simulate("click");
  // expect(callBack).toHaveBeenCalled();
  expect(spy).toHaveBeenCalled();
});
it("is cartCount updated?", () => {
  expect(appStore.cartItemList.length).toBe(1);
});
it("snapshot testing ", () => {
  const wrapper = shallow(
    <EachProduct key="2" appStore={appStore} eachItem={product} />
  );
  expect(wrapper).toMatchSnapshot();
});
