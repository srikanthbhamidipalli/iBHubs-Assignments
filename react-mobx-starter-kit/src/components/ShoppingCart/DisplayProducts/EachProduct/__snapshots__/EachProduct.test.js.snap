// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing  1`] = `
<styled.div>
  <styled.div>
    <img
      alt="productImage"
      src="https://react-shopping-cart-67954.firebaseapp.com/static/media/12064273040195392_1.2995d79a.jpg"
    />
    <styled.div>
      <styled.span>
        Free Shipping
      </styled.span>
    </styled.div>
  </styled.div>
  <br />
  <div>
    Cat Tee Black T-Shirt
  </div>
  <br />
  <styled.div />
  <br />
  <styled.div>
    <select
      id="select-list"
      onClick={[Function]}
    >
      <option
        key="select"
        value="select"
      >
        select Size
      </option>
      <option
        key="S"
        value="S"
      >
        S
      </option>
      <option
        key="XS"
        value="XS"
      >
        XS
      </option>
    </select>
  </styled.div>
  <div
    id="price"
  >
    $
    10.9
  </div>
  <div>
    or
     
    9 *1.21
  </div>
  <br />
  <styled.div
    onClick={[Function]}
  >
    <styled.span>
      Add to Cart
    </styled.span>
  </styled.div>
</styled.div>
`;
